priv/repo/migrations/20180426164341_create_hourly_downloads.exs

```diff
@@ -0,0 +1,56 @@
+defmodule Castle.Repo.Migrations.CreateHourlyDownloads do
+  use Ecto.Migration
+
+  def change do
+    create table(:hourly_downloads, primary_key: false) do
+      add :podcast_id, :integer, null: false
+      add :episode_id, :uuid, null: false
+      add :dtim, :utc_datetime, null: false
+      add :count, :integer, null: false
+    end
+
+    execute """
+    CREATE OR REPLACE FUNCTION create_hourly_downloads_partition() RETURNS trigger AS
+    $$
+      DECLARE
+        partition_start DATE;
+        partition_end DATE;
+        partition TEXT;
+      BEGIN
+        partition_start := DATE_TRUNC('MONTH', NEW.dtim);
+        partition_end := partition_start + INTERVAL '1 MONTH';
+        partition := 'hourly_downloads_' || to_char(NEW.dtim,'YYYYMM');
+        IF NOT EXISTS(SELECT relname FROM pg_class WHERE relname=partition) THEN
+          RAISE NOTICE 'A partition has been created %',partition;
+          EXECUTE 'CREATE TABLE ' || partition || ' (CHECK (dtim >= DATE ''' || partition_start || ''' AND dtim < DATE ''' || partition_end || '''), CONSTRAINT ' || partition || '_uniq UNIQUE (episode_id, dtim)) INHERITS (hourly_downloads);';
```

Comment:
So apparently in postgres, when doing an `ON CONFLICT` ... you can specify a unique-constraint, but not a unique-index.  So I added the unique dtim/episode here instead of as an index.

Not entirely sure on the trade-offs there ... from what i understand, postgres is actually using an index under the hood to do constraints.